### PR TITLE
Services api first

### PR DESCRIPTION
**Pull request by crucibles**
resolved conflicts; adding for login api;

### CODE REVIEW

**Comment by crucibles** [comment]
@asumandang  hey check my pull request xD

File: src/app/gen-profile/gen-profile.component.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class GenProfileComponent implements OnInit {
   }
 
   getUser(): void {
-    this.userService.getUserById("1")
+    this.userService.getUser("1")
```
Comment (asumandang) [inline src/app/gen-profile/gen-profile.component.ts]: Dummy user_id; must be changed to a variable, not a string

File: src/app/gen-sidetab/gen-sidetab.component.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ export class GenSidetabComponent implements OnInit {
    * @summary Obtains information of the current user
    */
   getUser(): void {
-    this.userService.getUserById("1")
+    this.userService.getUser("1")
```
Comment (asumandang) [inline src/app/gen-sidetab/gen-sidetab.component.ts]: Dummy user_id; must be changed to a variable, not a string

File: src/app/section.service.ts [hunk 3]
```diff
@@ -35,8 +35,16 @@ import {
...
+
+import { 
+  Section 
+} from './section';
+
+import {
+  Headers, 
+  RequestOptions 
+} from '@angular/http';
+import { HttpHeaders } from '@angular/common/http';
```
Comment (asumandang) [inline src/app/section.service.ts]: Change imports

File: src/app/section.service.ts [hunk 4]
```diff
@@ -126,10 +134,13 @@ export class SectionService {
    * @returns array of sections
    */
   getUserSections(user_id): Observable<Section[]> {
-    //AHJ: not yet done :3
-    const url = `${this.secUrl}`;
-    let params = new HttpParams().set('section', '11');
-    return this.http.get<Section[]>(url, { params }).pipe(
+      const url = this.secUrl;
+            
+      let params = new HttpParams().set('id', user_id);
```
Comment (asumandang) [inline src/app/section.service.ts]: Wow! Great job! :dagger:

File: src/app/user.service.ts [hunk 5]
```diff
@@ -35,6 +35,7 @@ import {
 import {
   Quest
 } from './quest'
+import { HttpParams } from '@angular/common/http';
```
Comment (asumandang) [inline src/app/user.service.ts]: Change import

**Pull request merged**